lib/api/apiUtils/rateLimit/client.js

```diff
@@ -65,6 +76,50 @@ class RateLimitClient {
             })));
         });
     }
+
+    /**
+     * @callback RateLimitClient~grantTokens
+     * @param {Error|null} err
+     * @param {number|undefined} granted - Number of tokens granted (0 if denied)
+     */
+
+    /**
+     * Request tokens from Redis with atomic GCRA enforcement
+     *
+     * This method atomically:
+     * 1. Evaluates GCRA for N tokens
+     * 2. Grants tokens if quota available
+     * 3. Advances Redis counter by granted tokens
+     *
+     * Used by token reservation system to request capacity in advance.
+     *
+     * @param {string} bucketName - Bucket name
+     * @param {number} requested - Number of tokens requested
+     * @param {number} interval - Interval per request in ms
+     * @param {number} burstCapacity - Burst capacity in ms
+     * @param {RateLimitClient~grantTokens} cb - Callback
+     */
+    grantTokens(bucketName, requested, interval, burstCapacity, cb) {
+        const key = `throttling:bucket:${bucketName}:rps`;
```

Comment:
Since the client is named RateLimitClient, would it be better to prefix keys with `ratelimit:`? I also find it more precise.